gtsfm/scene_optimizer.py

```diff
@@ -251,106 +207,64 @@ def create_computation_graph(
-    pre_ba_poses = []
-    for i in pre_ba_sfm_data.get_valid_camera_indices():
-        pre_ba_poses.append(pre_ba_sfm_data.get_camera(i).pose())
-
-    post_ba_poses = []
-    for i in post_ba_sfm_data.get_valid_camera_indices():
-        post_ba_poses.append(post_ba_sfm_data.get_camera(i).pose())
-
-    fig = plt.figure()
-    ax = fig.add_subplot(projection="3d")
-
-    if gt_pose_graph is not None:
-        # Select ground truth poses that correspond to pre-BA and post-BA estimated poses
-        # some may have been lost after pruning to largest connected component
-        corresponding_gt_poses = [gt_pose_graph[i] for i in pre_ba_sfm_data.get_valid_camera_indices()]
-
-        # ground truth is used as the reference
-        pre_ba_poses = comp_utils.align_poses_sim3(corresponding_gt_poses, copy.deepcopy(pre_ba_poses))
-        post_ba_poses = comp_utils.align_poses_sim3(corresponding_gt_poses, copy.deepcopy(post_ba_poses))
-        viz_utils.plot_poses_3d(gt_pose_graph, ax, center_marker_color="m", label_name="GT")
-
-        post_ba_pose_errors_dict = metric_utils.compute_pose_errors(
-            gt_wTi_list=corresponding_gt_poses, wTi_list=post_ba_poses
+    viz_graph_list = []
+    viz_graph_list.append(
+        dask.delayed(viz_utils.save_sfm_data_viz)(ba_input_graph, os.path.join(PLOT_PATH, "ba_input"))
+    )
+    viz_graph_list.append(
+        dask.delayed(viz_utils.save_sfm_data_viz)(ba_output_graph, os.path.join(PLOT_PATH, "results"))
+    )
+    viz_graph_list.append(
+        dask.delayed(viz_utils.save_camera_poses_viz)(
+            ba_input_graph, ba_output_graph, gt_pose_graph, os.path.join(PLOT_PATH, "results")
         )
-        print("post_ba_pose_errors_dict: ", post_ba_pose_errors_dict)
-
-    viz_utils.plot_poses_3d(pre_ba_poses, ax, center_marker_color="c", label_name="Pre-BA")
-    viz_utils.plot_poses_3d(post_ba_poses, ax, center_marker_color="k", label_name="Post-BA")
+    )
+    return viz_graph_list
 
-    ax.legend(loc="upper left")
-    viz_utils.set_axes_equal(ax)
 
-    # save the 3D plot in the original view
-    fig.savefig(os.path.join(folder_name, "poses_3d.png"))
+def save_gtsfm_data(image_graph: Delayed, ba_input_graph: Delayed, ba_output_graph: Delayed):
+    """Saves the Gtsfm data before and after bundle adjustment.
 
-    # save the BEV representation
-    default_camera_elevation = 100  # in metres above ground
-    ax.view_init(azim=0, elev=default_camera_elevation)
-    fig.savefig(os.path.join(folder_name, "poses_bev.png"))
+    Args:
+        image_graph: input image wrapped as Delayed objects.
+        ba_input_graph: GtsfmData input to bundle adjustment wrapped as Delayed.
+        ba_output_graph: GtsfmData output to bundle adjustment wrapped as Delayed.
 
-    plt.close(fig)
+    Returns:
+        A list of delayed objects after saving the input and outputs to bundle adjustment.
+    """
+    saving_graph_list = []
+    # Save a duplicate in REACT_RESULTS_PATH.
```

Comment:
remove this comment?

Reply:
lets make this a pure refactor PR, and fix this later